synapse/http/server.py

```diff
@@ -296,8 +298,14 @@ def _async_render(self, request):
         # here. If it throws an exception, that is handled by the wrapper
         # installed by @request_handler.
 
+        def _unquote(s):
+            if PY3:
+                return urllib.parse.unquote(s)
```

Comment:
why do we not utf-8 decode here?

Reply:
It's already decoded by `_get_handler_for_request`.